smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructuredMemberWriter.java

```diff
@@ -65,6 +70,173 @@ void writeMembers(TypeScriptWriter writer, Shape shape) {
         }
     }
 
+    void writeFilterSensitiveLog(TypeScriptWriter writer, String objectParam) {
+        writer.write("...$L,", objectParam);
+        for (MemberShape member : members) {
+            if (isMemberOverwriteRequired(member)) {
+                Shape memberTarget = model.expectShape(member.getTarget());
+                String memberName = getSanitizedMemberName(member);
+                String memberParam = String.format("%s.%s", objectParam, memberName);
+                writer.openBlock("...($1L.$2L && { $2L: ", "}),", objectParam, memberName, () -> {
+                    if (member.getMemberTrait(model, SensitiveTrait.class).isPresent()) {
+                        // member is Sensitive, hide the value.
+                        writer.write("SENSITIVE_STRING");
+                    } else if (memberTarget instanceof StructureShape) {
+                        writeStructureFilterSensitiveLog(writer, memberTarget, memberParam);
+                    } else if (memberTarget instanceof CollectionShape) {
+                        MemberShape collectionMember = ((CollectionShape) memberTarget).getMember();
+                        writeCollectionFilterSensitiveLog(writer, collectionMember, memberParam);
+                    } else if (memberTarget instanceof MapShape) {
+                        MemberShape mapMember = ((MapShape) memberTarget).getValue();
+                        writeMapFilterSensitiveLog(writer, mapMember, memberParam);
+                    }
+                });
+            }
+        }
+    }
+
+    /**
+     * Recursively writes filterSensitiveLog for StructureShape.
+     */
+    private void writeStructureFilterSensitiveLog(
+            TypeScriptWriter writer,
+            Shape structureTarget,
+            String structureParam
+    ) {
+        if (structureTarget.getMemberTrait(model, SensitiveTrait.class).isPresent()) {
```

Comment:
This can just use `hasTrait`, since the only pathing to this code has already confirmed this is a `StructureShape` and not a `MemberShape`.

Reply:
Updated in [commit](https://github.com/awslabs/smithy-typescript/pull/170/commits/d4dc6fb241a22dffcd0b86b3646657c19037d649)
Also updated memberTarget variable name in respective functions